ttnn/ttnn/operations/transformer.py

```diff
@@ -152,57 +152,67 @@ def _golden_function(input_tensor: ttnn.Tensor, **_):
 )
 
 
-def _rotary_embedding_validate_input_tensors(operation_name, input_tensor, cos_cache, sin_cache, *args, **kwargs):
-    ttnn.validate_input_tensor(
-        operation_name,
-        input_tensor,
-        ranks=(4,),
-        dtypes=(ttnn.bfloat16, ttnn.bfloat8_b),
-        layouts=(ttnn.TILE_LAYOUT,),
-        can_be_on_device=True,
-        can_be_on_cpu=False,
-    )
-    ttnn.validate_input_tensor(
-        operation_name,
-        cos_cache,
-        ranks=(4,),
-        dtypes=(ttnn.bfloat16, ttnn.bfloat8_b),
-        layouts=(ttnn.TILE_LAYOUT,),
-        can_be_on_device=True,
-        can_be_on_cpu=False,
-    )
-    ttnn.validate_input_tensor(
-        operation_name,
-        sin_cache,
-        ranks=(4,),
-        dtypes=(ttnn.bfloat16, ttnn.bfloat8_b),
-        layouts=(ttnn.TILE_LAYOUT,),
-        can_be_on_device=True,
-        can_be_on_cpu=False,
-    )
-
-
-@ttnn.register_operation(
-    name="ttnn.transformer.rotary_embedding",
-    validate_input_tensors=_rotary_embedding_validate_input_tensors,
+# TODO: add golden function
+_golden_function = None
+rotary_embedding = ttnn.register_operation(golden_function=_golden_function)(
+    ttnn._ttnn.operations.transformer.rotary_embedding
 )
-def rotary_embedding(
-    input_tensor: ttnn.Tensor,
-    cos_cache: ttnn.Tensor,
-    sin_cache: ttnn.Tensor,
-    token_index: int,
-    memory_config: ttnn.MemoryConfig,
-) -> ttnn.Tensor:
-    """
-
-    rotary_embedding(input_tensor: ttnn.Tensor, cos_cache: ttnn.Tensor, sin_cache: ttnn.Tensor, token_index: int, memory_config: MemoryConfig) -> ttnn.Tensor
 
-    Applies the rotary embedding to the input_tensor tensor using the cos_cache and sin_cache tensors.
-
-    """
-    return ttnn.experimental.tensor.rotary_embedding(
-        input_tensor, cos_cache, sin_cache, token_index, output_mem_config=memory_config
-    )
+# TODO: Move to schema validation in CPP
```

Comment:
Just delete this before you merge but otherwise everything else looks good to me.